flink-ml-lib/src/main/java/org/apache/flink/ml/common/param/HasOutputCols.java

```diff
@@ -19,15 +19,13 @@
 package org.apache.flink.ml.common.param;
 
 import org.apache.flink.ml.param.Param;
-import org.apache.flink.ml.param.ParamValidators;
 import org.apache.flink.ml.param.StringArrayParam;
 import org.apache.flink.ml.param.WithParams;
 
 /** Interface for the shared outputCols param. */
 public interface HasOutputCols<T> extends WithParams<T> {
     Param<String[]> OUTPUT_COLS =
-            new StringArrayParam(
-                    "outputCols", "Output column names.", null, ParamValidators.nonEmptyArray());
+            new StringArrayParam("outputCols", "Output column names.", new String[0]);
```

Comment:
Why do we make this change? Is the semantic still intuitive for all classes that use this parameter?

Reply:
Because the current implementation uses `HasOutputCols` to specify the column name of the generated tables. As we should allow users not to config this parameter, we have to make the default value of `HasOutputCols` a valid but empty value. 

It does change the semantic of `HasOutputCols`. I'll move this parameter to the `HasColNames` parameter mentioned in previous comments and keep `HasOutputCols` as it has been.